File: lib/domain/repositories/i_branch_repository.dart
import '../entities/branch.dart';

/// 分支仓库接口
abstract class IBranchRepository {
  /// 获取所有分支
  Future<List<Branch>> getAll();

  /// 根据仓库 ID 获取分支列表
  Future<List<Branch>> getByRepositoryId(String repositoryId);

  /// 根据 ID 获取分支
  Future<Branch?> getById(String id);

  /// 创建分支
  Future<Branch> create(Branch branch);

  /// 更新分支
  Future<Branch> update(Branch branch);

  /// 删除分支（软删除）
  Future<void> delete(String id);
}

File: web/src/domain/repositories/i-branch-repository.ts
import { Branch } from '../entities/branch';

export interface IBranchRepository {
  getAll(): Promise<Branch[]>;
  getByRepositoryId(repositoryId: string): Promise<Branch[]>;
  getById(id: string): Promise<Branch | null>;
  create(branch: Branch): Promise<Branch>;
  update(branch: Branch): Promise<Branch>;
  delete(id: string): Promise<void>;
  search(query: string): Promise<Branch[]>;
  clear(): Promise<void>;
}
